service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/MedicationStatementMapper.java

```diff
@@ -243,4 +257,23 @@ private String buildPriorPrescription(MedicationRequest medicationRequest) {
         }
         return StringUtils.EMPTY;
     }
+
+    private String buildParticipant(MedicationRequest medicationRequest) {
+        Predicate<Reference> isPractitioner = buildPredicateReferenceIsA(ResourceType.Practitioner);
+        Predicate<Reference> isPatient = buildPredicateReferenceIsA(ResourceType.Patient);
+        Predicate<Reference> isRelevant = isPractitioner.or(isPatient);
+
+        if (medicationRequest.hasRecorder() && medicationRequest.getRecorder().hasReference()) {
+            final var reference = medicationRequest.getRecorder();
+            if (reference.hasReferenceElement() && isRelevant.test(reference)) {
```

Comment:
From the earlier discussion, shouldn't we only map to participant when the recorder reference is a Practitioner.
And if the recorder we generate some text prepended to the PertinentInformation?
(from @mattd-kainos 's comments on the updated AC)

Reply:
Yes. Waiting to find out which pertinentInformation block it should be added to